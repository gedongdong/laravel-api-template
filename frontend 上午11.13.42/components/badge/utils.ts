import { PresetColorTypes } from '../_util/colors';

export function isPresetColor(color?: string): boolean {
  return (PresetColorTypes as any[]).indexOf(color) !== -1;
}
