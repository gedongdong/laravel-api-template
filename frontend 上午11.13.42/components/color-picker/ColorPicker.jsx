/* eslint-disable */
import PropTypes from '../_util/vue-types';
import { defaultConfigProvider } from '../config-provider';
import BaseMixin from '../_util/BaseMixin';
import Pickr from '@simonwep/pickr/dist/pickr.es5.min';
import Icon from '../icon';
import LocaleReceiver from '../locale-provider/LocaleReceiver';
import enUS from './locale/en_US';
import debounce from 'lodash-es/debounce';

import { getOptionProps, findDOMNode } from '../_util/props-util';
let colors = '#194d33';
export default {
  name: 'AColorPicker',
  mixins: [BaseMixin],
  inject: {
    configProvider: { default: () => defaultConfigProvider },
  },
  model: {
    prop: 'value',
    event: 'change.value', //为了支持v-model直接返回颜色字符串 所以用了自定义的事件,与pickr自带change事件进行区分
  },
  props: {
    prefixCls: String,
    defaultValue: String, //默认值
    config: PropTypes.object, //pickr配置
    value: String, //颜色值
    locale: PropTypes.object, //双语包
    colorRounded: Number, //颜色数值保留几位小数
    size: PropTypes.oneOf(['default', 'small', 'large']).def('default'), //尺寸
    getPopupContainer: Function, //指定渲染容器
    disabled: { type: Boolean, default: false }, //是否禁用
    format: String, //颜色格式设置
    alpha: { type: Boolean, default: false }, //是否开启透明通道
    hue: { type: Boolean, default: true }, //是否开启色彩预选
  },

  data() {
    return {
      colors,
      myOpen: false,
      pickr: null,
      i18n: enUS,
    };
  },
  watch: {
    'configProvider.locale.ColorPicker': {
      handler(val) {
        if (this.locale) return;
        this.i18n = val;
        this.reInitialize();
      },
    },
    locale(val) {
      this.i18n = val.ColorPicker || val.lang;
      this.reInitialize();
    },
    value(val) {
      this.setColor(val);
    },
    disabled(val) {
      this.pickr[val ? 'disable' : 'enable']();
    },
    config: {
      handler() {
        this.reInitialize();
      },
      deep: true,
    },
    format(val) {
      const type = val.toLocaleUpperCase();
      let res = this.pickr.setColorRepresentation(type);
      if (res) {
        this.pickr.applyColor();
      } else {
        throw new TypeError('format was invalid');
      }
    },
  },
  mounted() {
    if (this.locale) {
      this.i18n = this.locale.ColorPicker || this.locale.lang;
    }
    this.createPickr();
    this.eventsBinding();
  },
  unmounted() {
    this.pickr.destroyAndRemove();
  },
  methods: {
    reInitialize() {
      this.pickr.destroyAndRemove();
      const dom = document.createElement('div');
      dom.id = 'color-picker' + this._uid;
      const box = findDOMNode(this).querySelector('#color-picker-box' + this._uid);
      box.appendChild(dom);
      this.createPickr();
      this.eventsBinding();
    },
    setColor: debounce(function (val) {
      this.pickr.setColor(val);
    }, 1000),
    eventsBinding() {
      const pickrEvents = [
        'init',
        'hide',
        'show',
        'save',
        'clear',
        'change',
        'changestop',
        'cancel',
        'swatchselect',
      ];
      Object.keys(this.$listeners).forEach(event => {
        pickrEvents.includes(event) && this.pickr.on(event, this.$listeners[event]);
      });
    },
    createPickr() {
      const { getPopupContainer } = getOptionProps(this);
      const { getPopupContainer: getContextPopupContainer } = this.configProvider;
      const container = getPopupContainer || getContextPopupContainer;
      this.pickr = Pickr.create(
        Object.assign(
          {
            el: '#color-picker' + this._uid,
            container: (container && container(findDOMNode(this))) || document.body,
            theme: 'monolith', // or 'monolith', or 'nano'
            default: this.value || this.defaultValue || null, // 有默认颜色pickr才可以获取到_representation
            components: {
              // Main components
              preview: true,
              opacity: this.alpha,
              hue: this.hue,
              // Input / output Options
              interaction: {
                hex: true,
                rgba: true,
                input: true,
                clear: true,
                save: true,
              },
            },
          },
          this.config,
          { i18n: this.i18n },
        ),
      )
        .on('save', (color, instance) => {
          if (color) {
            let _representation = instance._representation || 'HEXA';
            color = color['to' + _representation]().toString(this.colorRounded || 0);
          }
          this.$emit('change.value', color || '');
        })
        .on('hide', () => {
          this.setState({ myOpen: false });
        });
    },
    handleOpenChange() {
      const open = !this.myOpen;
      this.setState({ myOpen: open });
      this.pickr[open ? 'show' : 'hide']();
      this.$emit('openChange', open);
    },
    getDefaultLocale() {
      const result = {
        ...enUS,
        ...this.$props.locale,
      };
      result.lang = {
        ...result.lang,
        ...(this.$props.locale || {}).lang,
      };
      return result;
    },
    renderColorPicker() {
      const { prefixCls: customizePrefixCls } = this.$props;
      const { getPrefixCls } = this.configProvider;
      const prefixCls = getPrefixCls('color-picker', customizePrefixCls);
      const { disabled } = getOptionProps(this);
      const classString = {
        [prefixCls]: true,
        [`${prefixCls}-open`]: this.myOpen,
        [`${prefixCls}-lg`]: this.size === 'large',
        [`${prefixCls}-sm`]: this.size === 'small',
        [`${prefixCls}-disabled`]: this.disabled,
      };
      return (
        <div class={classString} tabindex={disabled ? -1 : 0} onClick={this.handleOpenChange}>
          <div class={`${prefixCls}-selection`}>
            <div id={'color-picker-box' + this._uid}>
              <div id={'color-picker' + this._uid}></div>
            </div>
            <Icon type="down" class={`${prefixCls}-icon`} />
          </div>
        </div>
      );
    },
  },
  render() {
    return (
      <LocaleReceiver
        componentName="ColorPicker"
        defaultLocale={this.getDefaultLocale}
        scopedSlots={{ default: this.renderColorPicker }}
      />
    );
  },
};
