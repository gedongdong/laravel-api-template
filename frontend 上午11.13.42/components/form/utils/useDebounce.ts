import type { Ref } from 'vue';
import { shallowRef, watchEffect } from 'vue';

export default function useDebounce<T>(value: Ref<T[]>): Ref<T[]> {
  const cacheValue = shallowRef(value.value.slice());
  let timeout: any = null;
  watchEffect(() => {
    clearTimeout(timeout);
    timeout = setTimeout(
      () => {
        cacheValue.value = value.value;
      },
      value.value.length ? 0 : 10,
    );
  });

  return cacheValue;
}
