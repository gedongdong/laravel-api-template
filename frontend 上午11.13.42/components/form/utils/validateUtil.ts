import RawAsyncValidator from 'async-validator';
import { cloneVNode } from 'vue';
import { warning } from '../../vc-util/warning';

import { setValues } from './valueUtil';
import { defaultValidateMessages } from './messages';
import { isValidElement } from '../../_util/props-util';
import type { InternalNamePath, RuleError, RuleObject, ValidateOptions } from '../interface';

// Remove incorrect original ts define
const AsyncValidator: any = RawAsyncValidator;

/**
 * Replace with template.
 *   `I'm ${name}` + { name: 'bamboo' } = I'm bamboo
 */
function replaceMessage(template: string, kv: Record<string, string>): string {
  return template.replace(/\$\{\w+\}/g, (str: string) => {
    const key = str.slice(2, -1);
    return kv[key];
  });
}

async function validateRule(
  name: string,
  value: any,
  rule: RuleObject,
  options: ValidateOptions,
  messageVariables?: Record<string, string>,
): Promise<string[]> {
  const cloneRule = { ...rule };

  // Bug of `async-validator`
  delete (cloneRule as any).ruleIndex;
  delete (cloneRule as any).trigger;

  // We should special handle array validate
  let subRuleField: RuleObject = null;
  if (cloneRule && cloneRule.type === 'array' && cloneRule.defaultField) {
    subRuleField = cloneRule.defaultField;
    delete cloneRule.defaultField;
  }

  const validator = new AsyncValidator({
    [name]: [cloneRule],
  });

  const messages = setValues({}, defaultValidateMessages, options.validateMessages);
  validator.messages(messages);

  let result = [];

  try {
    await Promise.resolve(validator.validate({ [name]: value }, { ...options }));
  } catch (errObj) {
    if (errObj.errors) {
      result = errObj.errors.map(({ message }, index: number) =>
        // Wrap VueNode with `key`
        isValidElement(message) ? cloneVNode(message, { key: `error_${index}` }) : message,
      );
    } else {
      console.error(errObj);
      result = [(messages.default as () => string)()];
    }
  }

  if (!result.length && subRuleField) {
    const subResults: string[][] = await Promise.all(
      (value as any[]).map((subValue: any, i: number) =>
        validateRule(`${name}.${i}`, subValue, subRuleField, options, messageVariables),
      ),
    );

    return subResults.reduce((prev, errors) => [...prev, ...errors], []);
  }

  // Replace message with variables
  const kv = {
    ...(rule as Record<string, string | number>),
    name,
    enum: (rule.enum || []).join(', '),
    ...messageVariables,
  };

  const fillVariableResult = result.map(error => {
    if (typeof error === 'string') {
      return replaceMessage(error, kv);
    }
    return error;
  });

  return fillVariableResult;
}

/**
 * We use `async-validator` to validate the value.
 * But only check one value in a time to avoid namePath validate issue.
 */
export function validateRules(
  namePath: InternalNamePath,
  value: any,
  rules: RuleObject[],
  options: ValidateOptions,
  validateFirst: boolean | 'parallel',
  messageVariables?: Record<string, string>,
) {
  const name = namePath.join('.');

  // Fill rule with context
  const filledRules: RuleObject[] = rules
    .map((currentRule, ruleIndex) => {
      const originValidatorFunc = currentRule.validator;
      const cloneRule = {
        ...currentRule,
        ruleIndex,
      };

      // Replace validator if needed
      if (originValidatorFunc) {
        cloneRule.validator = (rule: RuleObject, val: any, callback: (error?: string) => void) => {
          let hasPromise = false;

          // Wrap callback only accept when promise not provided
          const wrappedCallback = (...args: string[]) => {
            // Wait a tick to make sure return type is a promise
            Promise.resolve().then(() => {
              warning(
                !hasPromise,
                'Your validator function has already return a promise. `callback` will be ignored.',
              );

              if (!hasPromise) {
                callback(...args);
              }
            });
          };

          // Get promise
          const promise = originValidatorFunc(rule, val, wrappedCallback);
          hasPromise =
            promise && typeof promise.then === 'function' && typeof promise.catch === 'function';

          /**
           * 1. Use promise as the first priority.
           * 2. If promise not exist, use callback with warning instead
           */
          warning(hasPromise, '`callback` is deprecated. Please return a promise instead.');

          if (hasPromise) {
            (promise as Promise<void>)
              .then(() => {
                callback();
              })
              .catch(err => {
                callback(err || ' ');
              });
          }
        };
      }

      return cloneRule;
    })
    .sort(({ warningOnly: w1, ruleIndex: i1 }, { warningOnly: w2, ruleIndex: i2 }) => {
      if (!!w1 === !!w2) {
        // Let keep origin order
        return i1 - i2;
      }

      if (w1) {
        return 1;
      }

      return -1;
    });

  // Do validate rules
  let summaryPromise: Promise<RuleError[]>;

  if (validateFirst === true) {
    // >>>>> Validate by serialization
    summaryPromise = new Promise(async (resolve, reject) => {
      /* eslint-disable no-await-in-loop */
      for (let i = 0; i < filledRules.length; i += 1) {
        const rule = filledRules[i];
        const errors = await validateRule(name, value, rule, options, messageVariables);
        if (errors.length) {
          reject([{ errors, rule }]);
          return;
        }
      }
      /* eslint-enable */

      resolve([]);
    });
  } else {
    // >>>>> Validate by parallel
    const rulePromises: Promise<RuleError>[] = filledRules.map(rule =>
      validateRule(name, value, rule, options, messageVariables).then(errors => ({ errors, rule })),
    );

    summaryPromise = (
      validateFirst ? finishOnFirstFailed(rulePromises) : finishOnAllFailed(rulePromises)
    ).then((errors: RuleError[]): RuleError[] | Promise<RuleError[]> => {
      // Always change to rejection for Field to catch
      return Promise.reject<RuleError[]>(errors);
    });
  }

  // Internal catch error to avoid console error log.
  summaryPromise.catch(e => e);

  return summaryPromise;
}

async function finishOnAllFailed(rulePromises: Promise<RuleError>[]): Promise<RuleError[]> {
  return Promise.all(rulePromises).then(
    (errorsList: RuleError[]): RuleError[] | Promise<RuleError[]> => {
      const errors: RuleError[] = [].concat(...errorsList);

      return errors;
    },
  );
}

async function finishOnFirstFailed(rulePromises: Promise<RuleError>[]): Promise<RuleError[]> {
  let count = 0;

  return new Promise(resolve => {
    rulePromises.forEach(promise => {
      promise.then(ruleError => {
        if (ruleError.errors.length) {
          resolve([ruleError]);
        }

        count += 1;
        if (count === rulePromises.length) {
          resolve([]);
        }
      });
    });
  });
}
