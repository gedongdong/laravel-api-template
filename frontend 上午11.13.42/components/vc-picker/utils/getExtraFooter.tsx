import type { PanelMode } from '../interface';

export default function getExtraFooter(
  prefixCls: string,
  mode: PanelMode,
  renderExtraFooter?: (mode: PanelMode) => any,
) {
  if (!renderExtraFooter) {
    return null;
  }

  return <div class={`${prefixCls}-footer-extra`}>{renderExtraFooter(mode)}</div>;
}
