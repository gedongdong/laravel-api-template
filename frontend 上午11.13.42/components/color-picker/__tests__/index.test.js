import { mount } from '@vue/test-utils';
import ColorPicker from '..';
import { asyncExpect } from '../../../tests/utils';
describe('ColorPicker', () => {
  xit('should support default value', async () => {
    const wrapper = mount(
      {
        render() {
          return <ColorPicker default-value="#cd0200" getPopupContainer={p => p}></ColorPicker>;
        },
      },
      { sync: false, attachTo: 'body' },
    );
    await asyncExpect(() => {
      expect(wrapper.html()).toMatchSnapshot();
      wrapper.unmount();
    }, 1000);
  });
  xit('should support v-model', async () => {
    let color = 'rgba(10, 10, 10, 1)';
    const wrapper = mount(
      {
        data() {
          return {
            color,
          };
        },
        render() {
          return <ColorPicker v-model={this.color} getPopupContainer={p => p}></ColorPicker>;
        },
        mounted() {
          this.color = 'rgba(110, 120, 130, 1)';
        },
      },
      { sync: false, attachTo: 'body' },
    );

    await asyncExpect(() => {
      expect(wrapper.html()).toMatchSnapshot();
      wrapper.unmount();
    }, 1000);
  });
  xit('should support disabled', async () => {
    const wrapper = mount(
      {
        data() {
          return {
            disabled: false,
          };
        },
        render() {
          return <ColorPicker disabled={this.disabled} getPopupContainer={p => p}></ColorPicker>;
        },
        mounted() {
          this.disabled = true;
        },
      },
      { sync: false, attachTo: 'body' },
    );

    await asyncExpect(async () => {
      expect(wrapper.html()).toMatchSnapshot();
      await asyncExpect(() => {
        wrapper.unmount();
      });
    }, 1000);
  });
  xit('should support format', async () => {
    const wrapper = mount(
      {
        data() {
          return {
            format: 'RGBA',
          };
        },
        render() {
          return <ColorPicker format={this.format} getPopupContainer={p => p}></ColorPicker>;
        },
        mounted() {
          this.format = 'HEX';
        },
      },
      { sync: false, attachTo: 'body' },
    );

    await asyncExpect(async () => {
      expect(wrapper.html()).toMatchSnapshot();
      await asyncExpect(() => {
        wrapper.unmount();
      });
    }, 1000);
  });
  xit('prop locale should works', async () => {
    const wrapper = mount(
      {
        data() {
          return {
            locale: {
              lang: {
                'btn:save': 'セーブ',
                'btn:cancel': 'キャンセル',
                'btn:clear': '晴れ',
              },
            },
          };
        },
        render() {
          return (
            <ColorPicker default-value="#cd0200" locale={this.locale} getPopupContainer={p => p} />
          );
        },
        mounted() {
          this.locale = {
            lang: {
              'btn:save': '1セーブ',
              'btn:cancel': '1キャンセル',
              'btn:clear': '1晴れ',
            },
          };
        },
      },
      { sync: false, attachTo: 'body' },
    );
    await asyncExpect(async () => {
      expect(wrapper.html()).toMatchSnapshot();
      await asyncExpect(() => {
        wrapper.unmount();
      });
    }, 1000);
  });
  xit('save event should works', async () => {
    const wrapper = mount(
      {
        render() {
          return (
            <ColorPicker default-value="#cd0200" getPopupContainer={p => p} onSave={this.save} />
          );
        },
        methods: {
          save(val) {
            return val;
          },
        },
      },
      { sync: false, attachTo: 'body' },
    );
    await asyncExpect(async () => {
      wrapper.find('.pcr-save').trigger('click');
      expect(wrapper.html()).toMatchSnapshot();
      await asyncExpect(() => {
        wrapper.unmount();
      });
    }, 1000);
  });
});
