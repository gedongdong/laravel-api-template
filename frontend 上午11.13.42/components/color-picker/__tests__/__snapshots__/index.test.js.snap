// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker prop locale should works 1`] = `
<div tabindex="0" class="ant-color-picker">
  <div class="ant-color-picker-selection">
    <div id="color-picker-box25">
      <div class="pickr">

        <button type="button" class="pcr-button" role="button" aria-label="toggle color picker dialog"></button>


      </div>
    </div><i aria-label="icon: down" class="anticon anticon-down ant-color-picker-icon"><svg viewBox="64 64 896 896" data-icon="down" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class="">
        <path d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"></path>
      </svg></i>
  </div>
  <div class="pcr-app" data-theme="monolith" aria-label="color picker dialog" role="window">
    <div class="pcr-selection">
      <div class="pcr-color-preview">
        <button type="button" class="pcr-last-color" aria-label="use previous color"></button>
        <div class="pcr-current-color"></div>
      </div>

      <div class="pcr-color-palette">
        <div class="pcr-picker"></div>
        <div class="pcr-palette" tabindex="0" aria-label="color selection area" role="listbox"></div>
      </div>

      <div class="pcr-color-chooser">
        <div class="pcr-picker"></div>
        <div class="pcr-hue pcr-slider" tabindex="0" aria-label="hue selection slider" role="slider"></div>
      </div>

      <div class="pcr-color-opacity" style="display:none" hidden="">
        <div class="pcr-picker"></div>
        <div class="pcr-opacity pcr-slider" tabindex="0" aria-label="selection slider" role="slider"></div>
      </div>
    </div>

    <div class="pcr-swatches "></div>

    <div class="pcr-interaction">
      <input class="pcr-result" type="text" spellcheck="false" aria-label="color input field">

      <input class="pcr-type active" data-type="HEXA" value="HEXA" type="button">
      <input class="pcr-type" data-type="RGBA" value="RGBA" type="button">
      <input class="pcr-type" data-type="HSLA" value="HSLA" type="button" style="display:none" hidden="">
      <input class="pcr-type" data-type="HSVA" value="HSVA" type="button" style="display:none" hidden="">
      <input class="pcr-type" data-type="CMYK" value="CMYK" type="button" style="display:none" hidden="">

      <input class="pcr-save" value="1セーブ" type="button" aria-label="save and close">
      <input class="pcr-cancel" value="1キャンセル" type="button" style="display:none" hidden="" aria-label="cancel and close">
      <input class="pcr-clear" value="1晴れ" type="button" aria-label="clear and close">
    </div>
  </div>
</div>
`;

exports[`ColorPicker save event should works 1`] = `
<div tabindex="0" class="ant-color-picker">
  <div class="ant-color-picker-selection">
    <div id="color-picker-box31">
      <div class="pickr">

        <button type="button" class="pcr-button" role="button" aria-label="toggle color picker dialog" style="color: rgb(0, 0, 0);"></button>


      </div>
    </div><i aria-label="icon: down" class="anticon anticon-down ant-color-picker-icon"><svg viewBox="64 64 896 896" data-icon="down" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class="">
        <path d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"></path>
      </svg></i>
  </div>
  <div class="pcr-app visible" data-theme="monolith" aria-label="color picker dialog" role="window" style="left: 512px; top: 384px;">
    <div class="pcr-selection">
      <div class="pcr-color-preview">
        <button type="button" class="pcr-last-color" aria-label="use previous color" style="color: rgb(0, 0, 0);"></button>
        <div class="pcr-current-color"></div>
      </div>

      <div class="pcr-color-palette">
        <div class="pcr-picker"></div>
        <div class="pcr-palette" tabindex="0" aria-label="color selection area" role="listbox"></div>
      </div>

      <div class="pcr-color-chooser">
        <div class="pcr-picker"></div>
        <div class="pcr-hue pcr-slider" tabindex="0" aria-label="hue selection slider" role="slider"></div>
      </div>

      <div class="pcr-color-opacity" style="display:none" hidden="">
        <div class="pcr-picker"></div>
        <div class="pcr-opacity pcr-slider" tabindex="0" aria-label="selection slider" role="slider"></div>
      </div>
    </div>

    <div class="pcr-swatches "></div>

    <div class="pcr-interaction">
      <input class="pcr-result" type="text" spellcheck="false" aria-label="color input field">

      <input class="pcr-type active" data-type="HEXA" value="HEXA" type="button">
      <input class="pcr-type" data-type="RGBA" value="RGBA" type="button">
      <input class="pcr-type" data-type="HSLA" value="HSLA" type="button" style="display:none" hidden="">
      <input class="pcr-type" data-type="HSVA" value="HSVA" type="button" style="display:none" hidden="">
      <input class="pcr-type" data-type="CMYK" value="CMYK" type="button" style="display:none" hidden="">

      <input class="pcr-save" value="Save" type="button" aria-label="save and close">
      <input class="pcr-cancel" value="Cancel" type="button" style="display:none" hidden="" aria-label="cancel and close">
      <input class="pcr-clear" value="Clear" type="button" aria-label="clear and close">
    </div>
  </div>
</div>
`;

exports[`ColorPicker should support default value 1`] = `
<div tabindex="0" class="ant-color-picker">
  <div class="ant-color-picker-selection">
    <div id="color-picker-box1">
      <div class="pickr">

        <button type="button" class="pcr-button" role="button" aria-label="toggle color picker dialog"></button>


      </div>
    </div><i aria-label="icon: down" class="anticon anticon-down ant-color-picker-icon"><svg viewBox="64 64 896 896" data-icon="down" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class="">
        <path d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"></path>
      </svg></i>
  </div>
  <div class="pcr-app" data-theme="monolith" aria-label="color picker dialog" role="window">
    <div class="pcr-selection">
      <div class="pcr-color-preview">
        <button type="button" class="pcr-last-color" aria-label="use previous color"></button>
        <div class="pcr-current-color"></div>
      </div>

      <div class="pcr-color-palette">
        <div class="pcr-picker"></div>
        <div class="pcr-palette" tabindex="0" aria-label="color selection area" role="listbox"></div>
      </div>

      <div class="pcr-color-chooser">
        <div class="pcr-picker"></div>
        <div class="pcr-hue pcr-slider" tabindex="0" aria-label="hue selection slider" role="slider"></div>
      </div>

      <div class="pcr-color-opacity" style="display:none" hidden="">
        <div class="pcr-picker"></div>
        <div class="pcr-opacity pcr-slider" tabindex="0" aria-label="selection slider" role="slider"></div>
      </div>
    </div>

    <div class="pcr-swatches "></div>

    <div class="pcr-interaction">
      <input class="pcr-result" type="text" spellcheck="false" aria-label="color input field">

      <input class="pcr-type active" data-type="HEXA" value="HEXA" type="button">
      <input class="pcr-type" data-type="RGBA" value="RGBA" type="button">
      <input class="pcr-type" data-type="HSLA" value="HSLA" type="button" style="display:none" hidden="">
      <input class="pcr-type" data-type="HSVA" value="HSVA" type="button" style="display:none" hidden="">
      <input class="pcr-type" data-type="CMYK" value="CMYK" type="button" style="display:none" hidden="">

      <input class="pcr-save" value="Save" type="button" aria-label="save and close">
      <input class="pcr-cancel" value="Cancel" type="button" style="display:none" hidden="" aria-label="cancel and close">
      <input class="pcr-clear" value="Clear" type="button" aria-label="clear and close">
    </div>
  </div>
</div>
`;

exports[`ColorPicker should support disabled 1`] = `
<div tabindex="-1" class="ant-color-picker ant-color-picker-disabled">
  <div class="ant-color-picker-selection">
    <div id="color-picker-box13">
      <div class="pickr">

        <button type="button" class="pcr-button disabled" role="button" aria-label="toggle color picker dialog"></button>


      </div>
    </div><i aria-label="icon: down" class="anticon anticon-down ant-color-picker-icon"><svg viewBox="64 64 896 896" data-icon="down" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class="">
        <path d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"></path>
      </svg></i>
  </div>
  <div class="pcr-app" data-theme="monolith" aria-label="color picker dialog" role="window">
    <div class="pcr-selection">
      <div class="pcr-color-preview">
        <button type="button" class="pcr-last-color" aria-label="use previous color"></button>
        <div class="pcr-current-color"></div>
      </div>

      <div class="pcr-color-palette">
        <div class="pcr-picker"></div>
        <div class="pcr-palette" tabindex="0" aria-label="color selection area" role="listbox"></div>
      </div>

      <div class="pcr-color-chooser">
        <div class="pcr-picker"></div>
        <div class="pcr-hue pcr-slider" tabindex="0" aria-label="hue selection slider" role="slider"></div>
      </div>

      <div class="pcr-color-opacity" style="display:none" hidden="">
        <div class="pcr-picker"></div>
        <div class="pcr-opacity pcr-slider" tabindex="0" aria-label="selection slider" role="slider"></div>
      </div>
    </div>

    <div class="pcr-swatches "></div>

    <div class="pcr-interaction">
      <input class="pcr-result" type="text" spellcheck="false" aria-label="color input field">

      <input class="pcr-type active" data-type="HEXA" value="HEXA" type="button">
      <input class="pcr-type" data-type="RGBA" value="RGBA" type="button">
      <input class="pcr-type" data-type="HSLA" value="HSLA" type="button" style="display:none" hidden="">
      <input class="pcr-type" data-type="HSVA" value="HSVA" type="button" style="display:none" hidden="">
      <input class="pcr-type" data-type="CMYK" value="CMYK" type="button" style="display:none" hidden="">

      <input class="pcr-save" value="Save" type="button" aria-label="save and close">
      <input class="pcr-cancel" value="Cancel" type="button" style="display:none" hidden="" aria-label="cancel and close">
      <input class="pcr-clear" value="Clear" type="button" aria-label="clear and close">
    </div>
  </div>
</div>
`;

exports[`ColorPicker should support format 1`] = `
<div tabindex="0" class="ant-color-picker ant-color-picker-open">
  <div class="ant-color-picker-selection">
    <div id="color-picker-box19">
      <div class="pickr">

        <button type="button" class="pcr-button" role="button" aria-label="toggle color picker dialog" style="color: rgb(0, 0, 0);"></button>


      </div>
    </div><i aria-label="icon: down" class="anticon anticon-down ant-color-picker-icon"><svg viewBox="64 64 896 896" data-icon="down" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class="">
        <path d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"></path>
      </svg></i>
  </div>
  <div class="pcr-app visible" data-theme="monolith" aria-label="color picker dialog" role="window" style="left: 512px; top: 384px;">
    <div class="pcr-selection">
      <div class="pcr-color-preview">
        <button type="button" class="pcr-last-color" aria-label="use previous color" style="color: rgb(0, 0, 0);"></button>
        <div class="pcr-current-color"></div>
      </div>

      <div class="pcr-color-palette">
        <div class="pcr-picker"></div>
        <div class="pcr-palette" tabindex="0" aria-label="color selection area" role="listbox"></div>
      </div>

      <div class="pcr-color-chooser">
        <div class="pcr-picker"></div>
        <div class="pcr-hue pcr-slider" tabindex="0" aria-label="hue selection slider" role="slider"></div>
      </div>

      <div class="pcr-color-opacity" style="display:none" hidden="">
        <div class="pcr-picker"></div>
        <div class="pcr-opacity pcr-slider" tabindex="0" aria-label="selection slider" role="slider"></div>
      </div>
    </div>

    <div class="pcr-swatches "></div>

    <div class="pcr-interaction">
      <input class="pcr-result" type="text" spellcheck="false" aria-label="color input field">

      <input class="pcr-type active" data-type="HEXA" value="HEXA" type="button">
      <input class="pcr-type" data-type="RGBA" value="RGBA" type="button">
      <input class="pcr-type" data-type="HSLA" value="HSLA" type="button" style="display:none" hidden="">
      <input class="pcr-type" data-type="HSVA" value="HSVA" type="button" style="display:none" hidden="">
      <input class="pcr-type" data-type="CMYK" value="CMYK" type="button" style="display:none" hidden="">

      <input class="pcr-save" value="Save" type="button" aria-label="save and close">
      <input class="pcr-cancel" value="Cancel" type="button" style="display:none" hidden="" aria-label="cancel and close">
      <input class="pcr-clear" value="Clear" type="button" aria-label="clear and close">
    </div>
  </div>
</div>
`;

exports[`ColorPicker should support v-model 1`] = `
<div tabindex="0" class="ant-color-picker">
  <div class="ant-color-picker-selection">
    <div id="color-picker-box7">
      <div class="pickr">

        <button type="button" class="pcr-button" role="button" aria-label="toggle color picker dialog"></button>


      </div>
    </div><i aria-label="icon: down" class="anticon anticon-down ant-color-picker-icon"><svg viewBox="64 64 896 896" data-icon="down" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class="">
        <path d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"></path>
      </svg></i>
  </div>
  <div class="pcr-app" data-theme="monolith" aria-label="color picker dialog" role="window">
    <div class="pcr-selection">
      <div class="pcr-color-preview">
        <button type="button" class="pcr-last-color" aria-label="use previous color"></button>
        <div class="pcr-current-color"></div>
      </div>

      <div class="pcr-color-palette">
        <div class="pcr-picker"></div>
        <div class="pcr-palette" tabindex="0" aria-label="color selection area" role="listbox"></div>
      </div>

      <div class="pcr-color-chooser">
        <div class="pcr-picker"></div>
        <div class="pcr-hue pcr-slider" tabindex="0" aria-label="hue selection slider" role="slider"></div>
      </div>

      <div class="pcr-color-opacity" style="display:none" hidden="">
        <div class="pcr-picker"></div>
        <div class="pcr-opacity pcr-slider" tabindex="0" aria-label="selection slider" role="slider"></div>
      </div>
    </div>

    <div class="pcr-swatches "></div>

    <div class="pcr-interaction">
      <input class="pcr-result" type="text" spellcheck="false" aria-label="color input field">

      <input class="pcr-type active" data-type="HEXA" value="HEXA" type="button">
      <input class="pcr-type" data-type="RGBA" value="RGBA" type="button">
      <input class="pcr-type" data-type="HSLA" value="HSLA" type="button" style="display:none" hidden="">
      <input class="pcr-type" data-type="HSVA" value="HSVA" type="button" style="display:none" hidden="">
      <input class="pcr-type" data-type="CMYK" value="CMYK" type="button" style="display:none" hidden="">

      <input class="pcr-save" value="Save" type="button" aria-label="save and close">
      <input class="pcr-cancel" value="Cancel" type="button" style="display:none" hidden="" aria-label="cancel and close">
      <input class="pcr-clear" value="Clear" type="button" aria-label="clear and close">
    </div>
  </div>
</div>
`;
